### PR TITLE
DataPrep Top Panel

### PR DESCRIPTION
**Pull request by elfenheart**
### Functionality:
- If no cookie present, the workspace modal should open
- User should be able to create, set, or delete a workspace
- If there is an active workspace, user can upload file and specify record delimiter (default to `\n`)

Currently pointing to `feature/ui-dataprep-improvement` branch for ease of review. Will change the merge branch to develop once that branch is merged

### CODE REVIEW

**Comment by tonybach** [comment]
Some observations while I was playing around:

- When I haven't enabled Data Prep yet, navigating to the page shows the content for a second before showing the 'Data Prep not enabled' message. Maybe we can show a spinning wheel instead?

dataprepstartup

- After I typed in the name of a workspace to create it, clicking on 'Set' accurately shows that the workspace does not exist yet. However, clicking on 'Delete' shows a success message. Is this a backend bug?

screen shot 2017-03-07 at 2 09 01 pm
screen shot 2017-03-07 at 2 08 33 pm

File: cdap-ui/app/cdap/components/DataPrep/TopPanel/WorkspaceModal.js [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+  }
+
+  setWorkspace() {
+    if (!this.state.workspaceId) { return; }
+    let workspaceId = this.state.workspaceId;
+    let namespace = NamespaceStore.getState().selectedNamespace;
+
+    let params = {
+      namespace,
+      workspaceId: workspaceId,
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/DataPrep/TopPanel/WorkspaceModal.js]: unnecessary property value?

File: cdap-ui/app/cdap/components/DataPrep/TopPanel/WorkspaceModal.js [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+          </div>
+        </ModalHeader>
+        <ModalBody>
+          {
+            this.state.activeWorkspace ? (
+              <div>
+                <h5>
+                  Current Active Workspace: <em>{this.state.activeWorkspace}</em>
+                </h5>
+                <hr/>
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/DataPrep/TopPanel/WorkspaceModal.js]: nit: space after hr

**Comment by elfenheart** [comment]
@tonybach the delete success message is a backend bug.


updated the PR. Will improve the loading animation position in subsequent PR

File: cdap-ui/app/cdap/components/DataPrep/TopPanel/WorkspaceModal.js [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+
+    let params = {
+      namespace,
+      workspaceId,
+      limit: 100
+    };
+
+    MyDataPrepApi.execute(params)
+      .subscribe((res) => {
+        cookie.save('DATAPREP_WORKSPACE', workspaceId);
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/DataPrep/TopPanel/WorkspaceModal.js]: Should we add `{path: '/'}` as option?

**Review by ajainarayanan — COMMENTED** [review]
Just minor comments.
- Saw a few console.log statements
- Saw a few places missing i18n
- Saw a few instances where we save value in cookie but don't have `path` option. Not sure if it needs to be consistent.

I am guessing first two will come in another PR. Other than that 👍 LGTM

**Comment by elfenheart** [comment]
console.log and i18n will come later on

**Review by ajainarayanan — APPROVED** [review]
👍

**Comment by elfenheart** [comment]
Closing in favor of https://github.com/caskdata/cdap/pull/8310

**Pull request closed without merging**